pkg/cmd/dockerregistry/dockerregistry.go

```diff
@@ -35,18 +40,104 @@ import (
 	_ "github.com/docker/distribution/registry/storage/driver/s3-aws"
 	_ "github.com/docker/distribution/registry/storage/driver/swift"
 
-	"strings"
+	kubeversion "k8s.io/kubernetes/pkg/version"
 
 	"github.com/openshift/origin/pkg/cmd/server/crypto"
 	"github.com/openshift/origin/pkg/cmd/util/clientcmd"
 	"github.com/openshift/origin/pkg/dockerregistry/server"
 	"github.com/openshift/origin/pkg/dockerregistry/server/api"
 	"github.com/openshift/origin/pkg/dockerregistry/server/audit"
 	registryconfig "github.com/openshift/origin/pkg/dockerregistry/server/configuration"
+	"github.com/openshift/origin/pkg/dockerregistry/server/prune"
+	"github.com/openshift/origin/pkg/version"
 )
 
+var pruneMode = flag.String("prune", "", "prune blobs from the storage and exit (check, delete)")
```

Comment:
I'd like to make `dockerregistry prune [-confirm]` and `dockerregistry serve [-config=<config.yml>]`, but it's insane to break backward compatibility and to change how the dockerregistry should be started. Also I don't want to handle case `dockerregistry -confirm`. So I end up with this solution.